raiden_contracts/utils/signature.py

```diff
@@ -8,12 +8,14 @@
 
 
 def sign(privkey: str, msg: bytes, v=0) -> bytes:
-    assert isinstance(msg, bytes)
-    assert isinstance(privkey, str)
+    if not isinstance(msg, bytes):
+        raise TypeError("sign(): msg is not an instance of bytes")
+    if len(msg) != 32:
+        raise ValueError("sign(): msg has to be exactly 32 bytes")
+    if not isinstance(privkey, str):
+        raise TypeError("sign(): privkey is not an instance of str")
 
```

Comment:
I think here's an `assert v in (0,27)` missing, otherwise looks good

Reply:
Also an assertion, that the signature is valid and recoverable would be nice

Reply:
I'll do these in this PR.